### PR TITLE
Remove `Ros2Value` encapsulation of `ArrayData`

### PR DESCRIPTION
**Pull request by haixuanTao**
Small refactoring:
- Remove Ros2Value encapsulation of `ArrayData` as it is no longer needed.
- Add a roundtrip test of copying `Arraydata` into a `Buffer` and back as I want to make sure that it works for all types and that we don't introduce any breaking bug.
- Fix clippy

### CODE REVIEW

File: examples/python-dataflow/run.rs
Comment (haixuanTao) [inline examples/python-dataflow/run.rs:1]: Exact Copy from `python-ros2-dataflow/run.rs`

**Review by phil-opp — APPROVED** [review]
Looks good, thanks a lot!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -122,7 +165,7 @@ jobs:
         env:
           QT_QPA_PLATFORM: offscreen
         run: |
-          source /opt/ros/humble/setup.bash && ros2 run turtlesim turtlesim_node &
+          source /opt/ros/humble/setup.bash && ros2 service call /reset std_srvs/srv/Empty &
```
[the remark this answers is not in the document]
Comment (haixuanTao) [reply .github/workflows/ci.yml:168]: Yep Sure!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,6 +56,7 @@ jobs:
       - name: "Build"
         run: cargo build --all
       - name: "Test"
+        if: runner.os == 'Linux' || runner.os == 'macOS'
```
[the remark this answers is not in the document]
Comment (haixuanTao) [reply .github/workflows/ci.yml:59]: So there is the weirdest `pdb` linking error in the world. See: https://github.com/dora-rs/dora/actions/runs/6455288058/job/17522513540  

I know the problem is not systemic as this CI/CD test passed: https://github.com/dora-rs/dora/actions/runs/6457648033/job/17531749038 

I tried to fix it by trying to maximize disk and memory capacity, but doesn't seems to make it work:
- By reducing the number of jobs in parallel: `cargo test --all --jobs 1`
- By removing docker image cache and Android on Windows.

So, I removed it but going to put a note on it.

[the remark this answers is not in the document]
Comment (phil-opp) [reply .github/workflows/ci.yml:59]: Maybe we can run at least library tests on Windows too? E.g. via `cargo test --all --lib`?

**Review by phil-opp — APPROVED** [review]
Thanks for the updates! Looks good to me, let's see whether the CI passes.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -122,14 +125,11 @@ jobs:
 
       # python examples
       - uses: actions/setup-python@v2
-        if: runner.os == 'Linux' || runner.os == 'macOS'
         with:
           python-version: "3.10"
       - name: "Python Dataflow example"
-        if: runner.os == 'Linux' || runner.os == 'macOS'
```
Comment (phil-opp) [inline .github/workflows/ci.yml:84]: Looks like this errors on Windows: https://github.com/dora-rs/dora/actions/runs/6531652388/job/17733278897?pr=359#step:14:32

Comment (haixuanTao) [reply .github/workflows/ci.yml]: Yeah, wanted to try it out, but it seems that the pip api is not the same on windows. I am going to remove this test from windows.

**Pull request merged**